sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -98,7 +98,9 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
       mirrors.distinct :+ "https://archive.apache.org/dist" :+ PROCESS_TABLES.releaseMirror
     logInfo(s"Trying to download Spark $version from $sites")
     for (site <- sites) {
-      val filename = if (version.startsWith("3")) {
+      val filename = if (version.startsWith("3.3")) {
```

Comment:
maybe we should make this more general: we don't want to update this whenever we update the Spark minor version like 3.4

Reply:
Yea, we can delete `hive-2.3` later.

Reply:
hmm seems this is not addressed, what if Spark 3.3 is released and `version` is 3.4, do we need update here again like:
```scala
val filename = if (version.startsWith("3.3") || version.startsWith("3.4")) {       
  s"spark-$version-bin-hadoop3.tgz"
} else if (version.startsWith("3")) {
  s"spark-$version-bin-hadoop3.2.tgz"
...
```

Reply:
> 

How about current?
```
      val filename = VersionUtils.majorMinorPatchVersion(version) match {
        case Some((major, _, _)) if major > 3 => s"spark-$version-bin-hadoop3.tgz"
        case Some((3, minor, _)) if minor >= 3 => s"spark-$version-bin-hadoop3.tgz"
        case Some((3, minor, _)) if minor < 3 => s"spark-$version-bin-hadoop3.2.tgz"
        case Some((_, _, _)) => s"spark-$version-bin-hadoop2.7.tgz"
      }
```